velox/common/file/FileSystems.cpp

```diff
@@ -94,10 +99,16 @@ class LocalFileSystem : public FileSystem {
       std::shared_ptr<FileSystem>(std::shared_ptr<const Config>)>
   fileSystemGenerator() {
     return [](std::shared_ptr<const Config> properties) {
-      return std::make_shared<LocalFileSystem>(properties);
+      // One instance of Local FileSystem is sufficient.
+      // Initialize on first access and reuse after that.
+      std::call_once(localFSRegistrationFlag, [&properties]() {
```

Comment:
A standard pattern is to use a static variable in the lambda, e.g. 

```
VectorFunctionMap& vectorFunctionFactories() {
  static VectorFunctionMap factories;
  return factories;
}
```

Would it make sense to use the same here?

Reply:
I pushed the static inside the lambda here. The pattern you proposed is somewhat similar to this change but not exact.

Reply:
I feel this should be sufficient.